aws/resource_aws_iam_policy.go

```diff
@@ -239,20 +240,31 @@ func resourceAwsIamPolicyDelete(d *schema.ResourceData, meta interface{}) error
 		return err
 	}
 
-	request := &iam.DeletePolicyInput{
-		PolicyArn: aws.String(d.Id()),
-	}
+	return resource.Retry(d.Timeout(schema.TimeoutDelete), func() *resource.RetryError {
```

Comment:
Was this change intended?

Reply:
When this functionality was in a separate extra resource, Terraform had trouble following the dependency and would attempt to destroying the managed policy before it was detached from the role. A retryable error allowed it to work by retrying after the detachment. I'll test to see if it works without this change now.

Reply:
Seems okay now.